File: 53.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        max_sum = nums[0]
        ending_here = nums[0]
        for i in range(1,len(nums)):
            
            if(nums[i] < ending_here + nums[i]):
                ending_here = ending_here + nums[i]
            else:
                ending_here  = nums[i]
            
                
            if(max_sum < ending_here):
                max_sum = ending_here
                    
                    
        return max_sum

File: 53.c


int maxSubArray(int* nums, int numsSize){
    int max = nums[0];
    int sum = 0;
    
    for(int i=0;i<numsSize;i++){
        sum = sum + nums[i];
        if(max < sum)
            max = sum;
        if(sum < 0)
            sum = 0;
    }
    return max;
}

int maxLength(int*nums , int numsSize){
	int max = nums[0];
    int sum = 0;
    int max_length = 0;
    int length = 0;
    for(int i=0;i<numsSize;i++){
        length++;
        sum = sum + nums[i];
        if(length >= max_length && sum >= max){
            max_length = length;
        }
        if(max < sum)
            max = sum;
        
        if(sum < 0){
            sum = 0;
            length = 0;
        }
    }
    printf("%d",max_length);
    return max_length;
	
}